whylog/teacher/__init__.py

```diff
@@ -1,22 +1,78 @@
+from whylog.teacher.constraint_links_base import ConstraintLinksBase
 from whylog.teacher.mock_outputs import create_sample_rule
 
 
+class PatternGroup(object):
+    """
+    Keeps "coordinates" of group that represents param in text
+
+    :param line_id: id of line to which group belongs.
+    :param number: number of group in line.
+                   Groups don't overlap. Numeration from left, from 1.
+    """
+
+    def __init__(self, line_id, group_number_in_line):
+        self.line_id = line_id
+        self.number = group_number_in_line
+
+
+class TeacherParser(object):
+    def __init__(self, line_object):
+        self.line = line_object
+        self.pattern_name = None
```

Comment:
those fields are never used. Remove them. You can add them back when they will be needed for something, but please try to keep dead code away from the master branch.